### PR TITLE
[5.8] Added last function

### PR DESCRIPTION
**Pull request by emre-kaya**
When we want to get last record, we use it;

```php
User::latest()->first();
// or
User::where('age', '>', 18)->latest()->first(); 
```

So i added `last` function, now we want to get last record, we use to `last` function

```php
User::last();
// or
User::where('age', '>', 18)->last();
```

### CODE REVIEW

File: src/Illuminate/Database/Concerns/BuildsQueries.php [hunk 1]
```diff
@@ -77,6 +77,17 @@ public function first($columns = ['*'])
         return $this->take(1)->get($columns)->first();
     }
 
+    /**
+     * Execut the query and get the last result.
```
Comment (staudenmeir) [inline src/Illuminate/Database/Concerns/BuildsQueries.php]: Typo: "Execute"

File: src/Illuminate/Database/Concerns/BuildsQueries.php [hunk 2]
```diff
@@ -77,6 +77,17 @@ public function first($columns = ['*'])
...
+     *
+     * @param array  $columns
+     * @return \Illuminate\Database\Eloquent\Model|object|static|null
+     */
+    public function last($columns = ['*'])
+    {
+        return $this->take(1)->latest()->get($columns)->first();
```
Comment (staudenmeir) [inline src/Illuminate/Database/Concerns/BuildsQueries.php]: You can use `return $this->latest()->first($columns);`.

**Comment by taylorotwell** [comment]
Just use lastest()->first() directly.

**Comment by musa11971** [comment]
Could this be reconsidered? Very useful imo.

**Comment by it-can** [comment]
this is very useful

**Comment by GrahamCampbell** [comment]
It is already possible using code that is almost the same amount of typing.

**Comment by musa11971** [comment]
@GrahamCampbell

@themsaid replied to a tweet by @calebporzio, stating that this could be re-opened if there’s enough demand. Which is the reason why this is gaining traction again, I guess.

Sure, it’s possible to do this already, but what’s wrong with have a convenience method? Laravel has plenty of those and it’s not like you’re forced to use it 😬

**Comment by driesvints** [comment]
Going to lock this as the PR was rejected, sorry.

**Pull request closed without merging**